packages/react-dom/src/test-utils/index.ts

```diff
@@ -8,7 +7,7 @@ declare global {
   var jest: { fn: typeof vi.fn } | undefined
 }
 
-const act = typeof React.act === 'function' ? React.act : DeprecatedReactTestUtils.act
+const act = React.act
```

Comment:
I found that act inside react-dom/test-utils has been deprecated. As mentioned in the documentation, I replaced it with the act from react.

https://react.dev/warnings/react-dom-test-utils